lib/fetchers/git.rb

```diff
@@ -38,10 +38,12 @@ def self.resolve(target, opts = {})
 
     def initialize(remote_url, opts = {})
       @branch = opts[:branch]
+      @profile_directory = nil
       @tag = opts[:tag]
       @ref = opts[:ref]
       @remote_url = remote_url
       @repo_directory = nil
+      @target_profile_path = opts[:profile_path]
```

Comment:
The rest of this PR has a good deal of conditional logic here.  I'm wondering if that conditional logic could be removed by making the _default_ profile_path `./`.  This of course depends on what behavior we want when someone happens to point at a git repository with multiple profiles.